src/controller/java/templates/helper.js

```diff
@@ -128,6 +146,94 @@ function asJniSignature(label, type)
   return templateUtil.templatePromise(this.global, promise)
 }
 
+function attributeCallbackTypeStringAsJniSignature(attributeTypeStr)
+{
+  const javaType = convertAttributeCallbackTypeStringToJavaType(attributeTypeStr);
+  switch (javaType) {
+  case 'Integer':
+    return 'I';
+  case 'Long':
+    return 'J';
+  case 'Boolean':
+    return 'Z';
+  case 'String':
+    return 'Ljava/lang/String;';
+  default:
+    error = 'Unhandled Java type ' + javaType;
+    throw error;
+  }
+}
+
+function asJavaAttributeCallbackType(attributeType)
+{
+  const attributeTypeStr = ChipHelper.asCallbackAttributeType(attributeType);
+  return convertAttributeCallbackTypeStringToJavaType(attributeTypeStr);
+}
+
+function convertAttributeCallbackTypeStringToJavaType(attributeTypeStr)
+{
+  switch (attributeTypeStr) {
+  case 'Int8u':
+  case 'Int16u':
+  case 'Int8s':
+  case 'Int16s':
+    return 'Integer';
+  case 'Int32u':
+  case 'Int64u':
```

Comment:
Is it not a problem that Int64u doesn't fit into a Java long because Java only has signed types?

Reply:
`uint64_t` can "fit" into a Java long, as long as we only use methods like [divideUnsigned](https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#divideUnsigned-int-int-). However, this is error-prone since in the current implementation there's no way to tell whether a signed or unsigned long is returned solely by looking at the code.

I looked into replacing `long` with `BigInteger`, but it seems like overkill to me, and I ran into some problems with the ZAP template using a promise in `{{#if}}` helpers.

I think a good middle ground here could be to create a type annotation to declare that a long field is unsigned. [Checker framework](https://checkerframework.org/manual/#signedness-checker) does something similar with [@Unsigned](https://checkerframework.org/api/org/checkerframework/checker/signedness/qual/Unsigned.html). We wouldn't have the compile-time checking that Checker has, unless we want to force clients to depend on Checker, but at least the signedness would be clear. WDYT?

Reply:
After some more thinking, this is probably out of scope for the PR, especially for the `BigInteger` solution, which would require a lot of changes in the template. I filed #8074 and added a TODO to track.